code/DynamicCache.php

```diff
@@ -397,14 +412,32 @@ public function run($url)
         // This is normally called in VersionedRequestFilter.
         Versioned::choose_site_stage();
 
+        // Get cache
```

Comment:
Since you've injected a new cache key on top of the cache key for the actual request, you've left the developer no ability to override or customise this key, in the same way that getCacheKey() previously allowed. This means you can no longer control segmentation based on url, and have assumed a fixed rule.

In fact nothing related to DB access should be necessary to build a key, since none of that work should be done at cache-check time.

If you need the ability to store this information, what you should do is implement a custom usercode-level cache, and use this api within your updateCacheKeyFragments. What you are doing is entirely possible without changes to the core module. Given that your fixed top level key is based on URL, and this URL is already passed to updateCacheKeyFragments, you shouldn't need more than minimal refactoring to add your own usercode level store.

Thus, I disagree that this change should be made a core feature of dynamic cache.